packages/core/src/browser/window/default-window-service.ts

```diff
@@ -38,7 +42,9 @@ export class DefaultWindowService implements WindowService, FrontendApplicationC
             if (!this.canUnload()) {
                 return this.preventUnload(event);
             }
+            this.fireUnload();
```

Comment:
```ts
if (!this.canUnload()) {
    this.preventUnload(event); // should be sync
}
// Will the browser unload?
if (!event.returnValue) {
    this.onUnloadEmitter.fire();
}
```

The idea here would be to fire `onUnload` in `beforeunload` always. This means we need to get rid of the firing in `preventUnload` on electron.

Reply:
Thanks! I am fine with your proposal, actually, I am fine with any proposal that works. Did you verify yours? If so, I am happy to apply the changes and proceed but if it works in theory only I would stick to mine.

Reply:
@marechal-p, let's put it this way; if you have a better, cleaner idea that works could you please submit a PR. I am more than happy to go through the verification steps and review it.

Reply:
Ok after experimenting I now understand that in a browser, once a user is prompted the confirmation window it means that we are out of the `beforeunload` handler, so the event can only be fired from `unload`. I did not now this before hand, it is my bad.

Reply:
> after experimenting

Thanks a lot for trying it out. 🙏  I do not like the changes, but I was not able to find another solution that works in both envs. I also tried to [integrate](https://github.com/eclipse-theia/theia/issues/8595#issuecomment-723470426) [`PageLifecycle.js`](https://github.com/GoogleChromeLabs/page-lifecycle), but it did not work for me 😕